src/twisted/python/_url.py

```diff
@@ -7,787 +7,7 @@
-        with the value C{value}, like so::
-
-            >>> URL.fromText(u'https://example.com/?x=y').add(u'x')
-            URL.fromText(u'https://example.com/?x=y&x')
-            >>> URL.fromText(u'https://example.com/?x=y').add(u'x', u'z')
-            URL.fromText(u'https://example.com/?x=y&x=z')
-
-        @param name: The name (the part before the C{=}) of the query parameter
-            to add.
-        @type name: L{unicode}
-
-        @param value: The value (the part after the C{=}) of the query
-            parameter to add.
-        @type value: L{unicode}
-
-        @return: a new L{URL} with the parameter added.
-        """
-        return self.replace(query=self.query + ((name, value),))
-
-
-    def set(self, name, value=None):
-        """
-        Create a new L{URL} with all existing occurrences of the query argument
-        C{name}, if any, removed, then add the argument with the given value,
-        like so::
-
-            >>> URL.fromText(u'https://example.com/?x=y').set(u'x')
-            URL.fromText(u'https://example.com/?x')
-            >>> URL.fromText(u'https://example.com/?x=y').set(u'x', u'z')
-            URL.fromText(u'https://example.com/?x=z')
-
-        @param name: The name (the part before the C{=}) of the query parameter
-            to add.
-        @type name: L{unicode}
-
-        @param value: The value (the part after the C{=}) of the query
-            parameter to add.
-        @type value: L{unicode}
-
-        @return: a new L{URL} with the parameter added or changed.
-        """
-        # Preserve the original position of the query key in the list
-        q = [(k, v) for (k, v) in self.query if k != name]
-        idx = next((i for (i, (k, v)) in enumerate(self.query)
-                    if k == name), -1)
-        q[idx:idx] = [(name, value)]
-        return self.replace(query=q)
-
-
-    def get(self, name):
-        """
-        Retrieve a list of values for the given named query parameter.
-
-        @param name: The name of the query parameter to retrieve.
-        @type name: L{unicode}
-
-        @return: all the values associated with the key; for example, for the
-            query string C{u"x=1&x=2"}, C{url.query.get(u"x")} would return
-            C{[u'1', u'2']}; C{url.query.get(u"y")} (since there is no C{"y"}
-            parameter) would return the empty list, C{[]}.
-        @rtype: L{list} of L{unicode}
-        """
-        return [value for (key, value) in self.query if name == key]
-
-
-    def remove(self, name):
-        """
-        Create a new L{URL} with all query arguments with the given name
-        removed.
-
-        @param name: The name of the query parameter to remove.
-        @type name: L{unicode}
-
-        @return: a new L{URL} with the parameter removed.
-        """
-        return self.replace(query=((k, v) for (k, v) in self.query
-                                   if k != name))
+__all__ = ["URL"]
```

Comment:
maybe we should add a comment here... in case someone will somehow direclty react this code and will don't know what happens... the comment  should describe that this is deprecated and hyperlink is used instead.